common/hash.go

```diff
@@ -0,0 +1,23 @@
+package common
+
+import (
+	"golang.org/x/crypto/blake2b"
+)
+
+func Blake2bHash(in []byte) (Hash, error) {
```

Comment:
We should probably use `trie/hash.go` and move it here

Reply:
nah i think we can leave this.. trie.Hash(node) is a specific case which takes a node and encodes then hashes the encoding, this function is a lot more general. alternatively i could change trie.Hash to use this func, what do you think